external_codes/boost_multi/multi/memory/allocator.hpp

```diff
@@ -49,7 +49,8 @@ class allocator{
 	bool operator!=(allocator const& o) const{return mp_ != o.mp_;}
 	using void_pointer = typename std::pointer_traits<decltype(std::declval<memory_type*>()->allocate(0, 0))>::template rebind<void>;
 	pointer allocate(size_type n){
-		return static_cast<pointer>(static_cast<void_pointer>(mp_->allocate(n*sizeof(value_type), alignof(T))));
+		return static_cast<pointer>(static_cast<void_pointer>(mp_->allocate(n*sizeof(value_type), 16)));
```

Comment:
why 16?

Reply:
This is a temporary fix until we update multi.
Cuda is incompatible with alignof() for std::complex<double>, so using alignof(T) in general
causes alignment issues when compiled in double precision for gpus.
With 16 all the regular types work fine. (Notice that cudamalloc aligns to 256).

Reply:
Should we update multi?